src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -0,0 +1,227 @@
+            # zip, happens often and indicates data for the given
+            # country - date is not available
+            logger.debug(
+                f"No zip data returned from url {url} "
+                f"check that the area and date exist."
+            )
+
+    return valid_file
+
+
+def _download_fewsnet_country(
+    date: datetime,
+    iso2: str,
+    output_dir: Path,
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers the iso2 country.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code for which the data should be downloaded
+    output_dir: Path
+        path of dir to which the zip content should be written
+    use_cache: bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    country_data : bool
+        if True, country data for the given date and iso2 exists
+    """
+    url_country_date = BASE_URL_COUNTRY.format(
+        iso2=iso2, YYYY=date.year, MM=date.month
+    )
+
+    output_dir_country = output_dir / f"{iso2}{date.strftime('%Y%m')}"
+    zip_filename = f"{output_dir_country.name}.zip"
+    if not output_dir_country.exists() or use_cache is False:
+        country_data = _download_zip(
+            url=url_country_date,
+            zip_filename=zip_filename,
+            output_dir=output_dir_country,
+        )
+    else:
+        country_data = True
+    return country_data
+
+
+def _download_fewsnet_region(
+    date: datetime,
+    region_name: str,
+    region_code: str,
+    output_dir: Path,
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers `region_name`.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    region_name : str
+        name of the region to which the `iso2` belongs,
+        e.g. "east-africa"
+    region_code : str
+        code that refers to the `region_name,
+        e.g. "EA"
+    output_dir : Path
+        path to dir to which the data should be written
+    use_cache : bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    region_data : bool
+        if True, region data for the given date and region name exists
+    """
+    url_region_date = BASE_URL_REGION.format(
+        region_code=region_code,
+        region_name=region_name,
+        YYYY=date.year,
+        MM=date.month,
+    )
+
+    output_dir_region = output_dir / f"{region_code}{date.strftime('%Y%m')}"
+    zip_filename_region = f"{output_dir_region.name}.zip"
+    if not output_dir_region.exists() or use_cache is False:
```

Comment:
Shouldn't this be `and`? (But if checking for the cache, shouldn't you check for the files themselves? Maybe it's tough with this data as you don't know the filenames)